### PR TITLE
Adds RawImageUploadSession API

### PR DESCRIPTION
**Pull request by haimasree**
Closes: #928

### CODE REVIEW

File: app/grandchallenge/api/urls.py [hunk 1]
```diff
@@ -25,6 +28,11 @@
 router.register(r"patients", PatientViewSet, basename="patient")
 router.register(r"studies", StudyViewSet, basename="study")
 router.register(r"worklists", WorklistViewSet, basename="worklist")
+router.register(
+    r"cases/upload-sessions",
+    RawImageUploadSessionViewSet,
+    basename="upload-sessions",
```
Comment (jmsmkn) [inline app/grandchallenge/api/urls.py]: Basename should be `upload-session`

File: app/grandchallenge/cases/models.py [hunk 2]
```diff
@@ -128,6 +128,10 @@ def get_absolute_url(self):
             "cases:raw-files-session-detail", kwargs={"pk": self.pk}
         )
 
+    @property
+    def api_url(self):
+        return reverse("api:upload-sessions-detail", kwargs={"pk": self.pk})
```
Comment (jmsmkn) [inline app/grandchallenge/cases/models.py]: Should be `"api:upload-session-detail"`

File: app/grandchallenge/cases/views.py [hunk 3]
```diff
@@ -1,8 +1,15 @@
 from django.http import Http404
 from django.views.generic import CreateView, DetailView
 from rest_framework.permissions import DjangoObjectPermissions
-from rest_framework.viewsets import ReadOnlyModelViewSet
+from rest_framework.viewsets import ReadOnlyModelViewSet, GenericViewSet
 from rest_framework_guardian.filters import DjangoObjectPermissionsFilter
+from rest_framework.viewsets import ReadOnlyModelViewSet, GenericViewSet
```
Comment (jmsmkn) [inline app/grandchallenge/cases/views.py]: This line is a duplicate of line 4 and should be removed

File: app/tests/cases_tests/factories.py [hunk 4]
```diff
@@ -152,3 +152,8 @@ def files(self, create, extracted, **kwargs):
         if create and not extracted:
             ImageFileFactoryWithMHDFile3DLarge4Slices(image=self)
             ImageFileFactoryWithRAWFile3DLarge4Slices(image=self)
+
+
+class RawImageUploadSessionFactory(factory.DjangoModelFactory):
```
Comment (jmsmkn) [inline app/tests/cases_tests/factories.py]: The subfactory for algorithm (images) is missing

File: app/tests/cases_tests/test_api.py [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+@pytest.mark.django_db
+def test_upload_session_list(client):
+    upload_session_1, upload_session_2 = (
+        RawImageUploadSessionFactory(),
+        RawImageUploadSessionFactory(),
+    )
+
+    user = UserFactory(is_staff=True)
+    response = get_view_for_user(
+        viewname="api:upload-sessions-list", user=user, client=client
```
Comment (jmsmkn) [inline app/tests/cases_tests/test_api.py]: `"api:upload-session-list"`

File: app/tests/cases_tests/test_api.py [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+    )
+    assert response.status_code == 200
+    assert response.json()["count"] == 2
+
+    response = get_view_for_user(
+        viewname="api:upload-sessions-detail",
```
Comment (jmsmkn) [inline app/tests/cases_tests/test_api.py]: `"api:upload-session-detail"`

File: app/tests/cases_tests/factories.py (hunk 4 above)
Comment (haimasree) [reply app/tests/cases_tests/factories.py]: Sorry I dont understand.

Comment (jmsmkn) [reply app/tests/cases_tests/factories.py]: Ok, in this case it doesn't matter a huge amount, we can add it later

**Pull request merged**